qcodes/instrument_drivers/tektronix/AWG5014.py

```diff
@@ -310,359 +366,428 @@ def __init__(self, name, setup_folder, address, reset=False,
                     vals=vals.Numbers(-2.7, 2.7),
                     get_parser=float)
 
-        # # Add functions
-
-        # self.add_function('get_state')
-        # self.add_function('set_event_jump_timing')
-        # self.add_function('get_event_jump_timing')
-        # self.add_function('generate_awg_file')
-        # self.add_function('send_awg_file')
-        # self.add_function('load_awg_file')
-        # self.add_function('get_error')
-        # self.add_function('pack_waveform')
-        # self.add_function('clear_visa')
-        # self.add_function('initialize_dc_waveforms')
-
-        # # Setup filepaths
-        self.waveform_folder = "Waveforms"
-        self._rem_file_path = "Z:\\Waveforms\\"
-
-        # NOTE! this directory has to exist on the AWG!!
-        self._setup_folder = setup_folder
-
-        self.goto_root()
-        self.change_folder(self.waveform_folder)
-
         self.set('trigger_impedance', 50)
         if self.get('clock_freq') != 1e9:
-            logging.warning('AWG clock freq not set to 1GHz')
+            log.warning('AWG clock freq not set to 1GHz')
 
         self.connect_message()
 
+    # Convenience parser
+    def newlinestripper(self, string):
+            if string.endswith('\n'):
+                return string[:-1]
+            else:
+                return string
+
     # Functions
     def get_all(self, update=True):
+        """
+        Deprecated function. Please don't use.
+
+        Function to get a snapshot of the state of all parameters and
+        functions of the instrument.
+        Note: methods of the Tektronix_AWG5014 class are not included.
+
+        Args:
+            update (bool): whether to return an updated state. Default: True
+
+        Returns:
+
+            dict: a JSON-serialisable dict with all information.
+
+        Raises:
+            DeprecationWarning
+        """
+        raise DeprecationWarning("Use snapshot(update=update) directly")
```

Comment:
this should be warnings.warn rather than raise. Otherwise the line below is never executed